extensions/compose/src/extension.ts

```diff
@@ -216,56 +214,68 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
   // Push the CLI tool as well (but it will do it postActivation so it does not block the activate() function)
   // Post activation
   setTimeout(() => {
-    postActivate(extensionContext, composeDownload).catch((error: unknown) => {
+    postActivate(composeDownload, detect).catch((error: unknown) => {
       console.error('Error activating extension', error);
     });
   }, 0);
 }
 
 // Activate the CLI tool (check version, etc) and register the CLi so it does not block activation.
-async function postActivate(
-  extensionContext: extensionApi.ExtensionContext,
-  composeDownload: ComposeDownload,
-): Promise<void> {
-  // The location of the binary (local storage folder)
-  const binaryPath = path.join(
-    extensionContext.storagePath,
-    'bin',
-    os.isWindows() ? composeCliName + '.exe' : composeCliName,
-  );
-  let binaryVersion = '';
-
-  // Retrieve the version of the binary by running exec with --short
-  try {
-    const result = await extensionApi.process.exec(binaryPath, ['--version', '--short']);
-    binaryVersion = result.stdout;
-  } catch (e) {
-    console.error(`Error getting compose version: ${e}`);
+async function postActivate(composeDownload: ComposeDownload, detect: Detect): Promise<void> {
+  // build executable name for current platform
+  const executable = os.isWindows() ? composeCliName + '.exe' : composeCliName;
+
+  // binary info
+  let binaryInfo: { version: string; path: string; updatable?: boolean } | undefined = undefined;
+
+  // let's check for system-wide
+  const installedSystemWide = await detect.checkSystemWideDockerCompose();
+  if (installedSystemWide) {
+    binaryInfo = await detect.getDockerComposeBinaryInfo(executable);
+  } else {
+    // if not installed, let's check for local version
+    const extensionExecutable = await detect.getStoragePath();
+    // if local version exists
+    if (extensionExecutable.length !== 0) {
+      binaryInfo = await detect.getDockerComposeBinaryInfo(executable, detect.getExtensionStorageBin());
+    }
   }
 
-  // Register the CLI tool so it appears in the preferences page. We will detect which version is being ran by
-  // checking the local storage folder for the binary. If it exists, we will run `--version` and parse the information.
+  // if no binary detected let's just stop here
+  if (!binaryInfo) return;
```

Comment:
If compose is not installed, this removes it from the Settings entirely, i.e. no trace that Podman Desktop even knows what compose is. If it is not installed it should still be in the Settings page, show that it's not installed, and ideally install it.

Reply:
https://github.com/containers/podman-desktop/assets/42176370/6e0df8b7-2f8d-4f73-bfdd-85005f30648f

We do not have a status for not-installed in the CLI tool, we need to provide some required arguments, 

https://github.com/containers/podman-desktop/blob/0fb809c40babc26327022fbfe3ce2bb5b9d7d6d2/packages/extension-api/src/extension-api.d.ts#L4269-L4284

You need to use the setup workflow to have it install

Reply:
> I don't think CLI tools should be under both Resources and CLI Tools

This is not a problem related to the issue here. Basically the way the extension api is made, the extension should register _an existing and valid_ version, and it accepts an update callback.

The current design does not support registering an installation, and here it is in the Resources because it is part of the onBoarding workflow of the Compose extension.

> My concern here was that before this change I can see compose listed under CLI Tools even if I don't have it installed yet, but after this PR it's not listed at all.

Even if it was listed, you could not install it, because it was not suppose to be here, leading to user being perplex has to understand if it was or not installed. 

> We might need a separate issue to show 'not installed' state better

I agree, that we should have a setup/install option here, it would simplify a lot.

> but IMHO it should remain visible

Not in this PR.

Reply:
As discussed on the standup call I don't like this 'regression' of compose not appearing in CLI Tools anymore if you don't have it installed, but I'm ok with this as a temporary state with #8003 open to track longer term behaviour. Thanks. 👍🏼